flake.nix

```diff
@@ -86,7 +87,14 @@
       in
       {
         defaultPackage = formatter; # or replace it with your desired default package.
-        devShell = pkgs.mkShell { buildInputs = devShellPackages; };
+        devShell = pkgs.mkShell {
+        buildInputs = devShellPackages;
+        shellHook = ''
+          echo "Welcome to the nix shell development environment!"
```

Comment:
Ah you didn't like my message :laughing: 

Maybe I should make this better, but I added this to figure out what `nix-shell` vs `nix shell` vs `nix develop` is. 

`nix shell` does not run shellHooks, so it will not work. I guess the proper way is just to echo the env var and see if it was set.